flink-yarn/src/test/java/org/apache/flink/yarn/FlinkYarnSessionCliTest.java

```diff
@@ -0,0 +1,293 @@
+		String[] params =
+			new String[] {"-yn", "2", "-ys", "3"};
+
+		FlinkYarnSessionCli yarnCLI = new FlinkYarnSessionCli("y", "yarn");
+
+		final CommandLine commandLine = yarnCLI.parseCommandLineOptions(params, true);
+
+		final Configuration configuration = new Configuration();
+
+		AbstractYarnClusterDescriptor descriptor = yarnCLI.createDescriptor(
+			configuration,
+			tmp.getRoot().getAbsolutePath(),
+			"",
+			commandLine);
+
+		final ClusterSpecification clusterSpecification = yarnCLI.createClusterSpecification(
+			configuration,
+			commandLine);
+
+		// each task manager has 3 slots but the parallelism is 7. Thus the slots should be increased.
+		assertEquals(3, clusterSpecification.getSlotsPerTaskManager());
+		assertEquals(2, clusterSpecification.getNumberTaskManagers());
+	}
+
+	@Test
+	public void testZookeeperNamespaceProperty() throws Exception {
+		String zkNamespaceCliInput = "flink_test_namespace";
+
+		String[] params = new String[] {"-yn", "2", "-yz", zkNamespaceCliInput};
+
+		FlinkYarnSessionCli yarnCLI = new FlinkYarnSessionCli("y", "yarn");
+
+		CommandLine commandLine = yarnCLI.parseCommandLineOptions(params, true);
+
+		AbstractYarnClusterDescriptor descriptor = yarnCLI.createDescriptor(
+			new Configuration(),
+			tmp.getRoot().getAbsolutePath(),
+			"",
+			commandLine);
+
+		assertEquals(zkNamespaceCliInput, descriptor.getZookeeperNamespace());
+	}
+
+	/**
+	 * Test that the CliFrontend is able to pick up the .yarn-properties file from a specified location.
+	 */
+	@Test
+	public void testResumeFromYarnPropertiesFile() throws Exception {
+
+		File directoryPath = writeYarnPropertiesFile(validPropertiesFile);
+
+		final FlinkYarnSessionCli flinkYarnSessionCli = new FlinkYarnSessionCli("y", "yarn");
+
+		final Configuration configuration = new Configuration();
+		configuration.setString(YarnConfigOptions.PROPERTIES_FILE_LOCATION, directoryPath.getAbsolutePath());
+
+		final CommandLine commandLine = flinkYarnSessionCli.parseCommandLineOptions(new String[] {}, true);
+
+		final String clusterId = flinkYarnSessionCli.getClusterId(
+			configuration,
+			commandLine);
+
+		assertEquals(TEST_YARN_APPLICATION_ID.toString(), clusterId);
+	}
+
+	@Test(expected = IllegalConfigurationException.class)
+	public void testInvalidYarnPropertiesFile() throws Exception {
+
+		File directoryPath = writeYarnPropertiesFile(invalidPropertiesFile);
+
+		final FlinkYarnSessionCli flinkYarnSessionCli = new FlinkYarnSessionCli("y", "yarn");
+
+		final Configuration configuration = new Configuration();
+		configuration.setString(YarnConfigOptions.PROPERTIES_FILE_LOCATION, directoryPath.getAbsolutePath());
+
+		final CommandLine commandLine = flinkYarnSessionCli.parseCommandLineOptions(new String[] {}, true);
+
+		flinkYarnSessionCli.getClusterId(
+			configuration,
+			commandLine);
+
+		fail("We should have failed reading the yarn properties file when retrieving the cluster id.");
```

Comment:
It is wrong to put `fail()` if your test expects an exception to be thrown. For example: 
```
@Test(expected = IllegalConfigurationException.class)
	public void testInvalidYarnPropertiesFile() throws Exception {
		try {
			File directoryPath = writeYarnPropertiesFile(invalidPropertiesFile);

			final FlinkYarnSessionCli flinkYarnSessionCli = new FlinkYarnSessionCli("y", "yarn");

			final Configuration configuration = new Configuration();
			configuration.setString(YarnConfigOptions.PROPERTIES_FILE_LOCATION, directoryPath.getAbsolutePath());

			final CommandLine commandLine = flinkYarnSessionCli.parseCommandLineOptions(new String[]{}, true);

			flinkYarnSessionCli.getClusterId(
				configuration,
				commandLine);
		} catch (final Exception e) {

		}

		fail("We should have failed reading the yarn properties file when retrieving the cluster id.");
	}
```

fails with 
```
java.lang.Exception: Unexpected exception, expected<org.apache.flink.configuration.IllegalConfigurationException> but was<java.lang.AssertionError>
```

The message *We should have failed reading the yarn properties file when retrieving the cluster id.* is lost.

Reply:
True. Will change it.